UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs

```diff
@@ -102,11 +104,18 @@ private void OnDisable()
 		IsInitialized = false;
 	}
 
-	public static Matrix MakeNewMatrix()
+	public static Matrix MakeNewMatrix(string Name = "Matrix")
 	{
-		//hummmm How to add matrix fluff?
+
+		if (string.IsNullOrEmpty(Name))
+		{
+			Name = "Matrix";
```

Comment:
what happens if there are multiple matricies that go by the name "Matrix"?

Reply:
it seems to be fine since everything uses the matrix ID (is int)